File: redwood-yoga-java/src/main/java/io/github/orioncraftmc/meditate/internal/enums/YGPrintOptions.java
package io.github.orioncraftmc.meditate.internal.enums;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public enum YGPrintOptions {
    YGPrintOptionsLayout(1),
    YGPrintOptionsStyle(2),
    YGPrintOptionsChildren(4);

    private final int value;

    YGPrintOptions(int value) {
        this.value = value;
    }

    public static @Nullable YGPrintOptions forValue(int value) {
        for (@NotNull YGPrintOptions options : values()) {
            if (options.value == value) {
                return options;
            }
        }

        return null;
    }

    public int getValue() {
        return value;
    }
}

File: redwood-yoga/src/commonMain/kotlin/app/cash/redwood/yoga/internal/enums/YGPrintOptions.kt
package app.cash.redwood.yoga.internal.enums

enum class YGPrintOptions {
  YGPrintOptionsLayout,
  YGPrintOptionsStyle,
  YGPrintOptionsChildren;

  fun getValue(): Int {
    return ordinal
  }

  companion object {
    fun forValue(value: Int): YGPrintOptions {
      return values()[value]
    }
  }
}
